app/javascript/mastodon/features/video/index.tsx

```diff
@@ -0,0 +1,1022 @@
+                role='slider'
+                aria-valuemin={0}
+                aria-valuenow={effectivelyMuted ? 0 : volume * 100}
+                aria-valuemax={100}
+                onMouseDown={handleVolumeMouseDown}
+                ref={volumeRef}
+                tabIndex={0}
+              >
+                <animated.div
+                  className='video-player__volume__current'
+                  style={{ width: style.volume }}
+                />
+
+                <animated.span
+                  className={classNames('video-player__volume__handle')}
+                  style={{ left: style.volume }}
+                />
+              </div>
+
+              {(detailed || fullscreen) && (
+                <span className='video-player__time'>
+                  <span className='video-player__time-current'>
+                    {formatTime(Math.floor(currentTime))}
+                  </span>
+                  <span className='video-player__time-sep'>/</span>
+                  <span className='video-player__time-total'>
+                    {formatTime(Math.floor(duration))}
+                  </span>
+                </span>
+              )}
+            </div>
+
+            <div className='video-player__buttons right'>
+              {!fullscreen && onOpenVideo && (
+                <button
+                  type='button'
+                  title={intl.formatMessage(messages.expand)}
+                  aria-label={intl.formatMessage(messages.expand)}
+                  className='player-button'
+                  onClick={handleOpenVideo}
+                >
+                  <Icon id='expand' icon={RectangleIcon} />
+                </button>
+              )}
+              {onCloseVideo && (
+                <button
+                  type='button'
+                  title={intl.formatMessage(messages.close)}
+                  aria-label={intl.formatMessage(messages.close)}
+                  className='player-button'
+                  onClick={handleCloseVideo}
+                >
+                  <Icon id='compress' icon={FullscreenExitIcon} />
+                </button>
+              )}
+              <button
+                type='button'
+                title={intl.formatMessage(
+                  fullscreen ? messages.exit_fullscreen : messages.fullscreen,
+                )}
+                aria-label={intl.formatMessage(
+                  fullscreen ? messages.exit_fullscreen : messages.fullscreen,
+                )}
+                className='player-button'
+                onClick={toggleFullscreen}
+              >
+                <Icon
+                  id={fullscreen ? 'compress' : 'arrows-alt'}
+                  icon={fullscreen ? FullscreenExitIcon : FullscreenIcon}
+                />
+              </button>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+// eslint-disable-next-line import/no-default-export
+export default Video;
```

Comment:
nit: does this still need to export a default? it doesn't seem to be imported like this anywhere.

Reply:
It's imported as an async component in a few places, which (at least as written) relies on default export. See e.g. `app/javascript/mastodon/features/ui/util/async-components.js`